ladybug/designday.py

```diff
@@ -0,0 +1,1109 @@
+from .location import Location
+from .futil import write_to_file
+from .dt import DateTime
+from .header import Header
+from .datacollection import DataCollection
+from .datatype import DataPoint
+from .analysisperiod import AnalysisPeriod
+from .sunpath import Sunpath
+
+from .skymodel import ashrae_revised_clear_sky
+from .skymodel import ashrae_clear_sky
+from .skymodel import energyplus_horizontal_infrared
+
+from .psychrometrics import rel_humid_from_db_dpt
+from .psychrometrics import dew_point_from_db_hr
+from .psychrometrics import dew_point_from_db_enth
+from .psychrometrics import dew_point_from_db_wb
+
+import math
+import os
+import re
+import codecs
+import platform
+
+day_types = ['SummerDesignDay', 'WinterDesignDay', 'Sunday', 'Monday',
+             'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Holiday',
+             'CustomDay1', 'CustomDay2']
+h_types = ['Wetbulb', 'Dewpoint', 'HumidityRatio', 'Enthalpy']
+sky_types = ['ASHRAEClearSky', 'ASHRAETau', 'ZhangHuang', 'Schedule']
+comments = ['!- Name',
+            '!- Month',
+            '!- Day of Month',
+            '!- Day Type',
+            '!- Max Dry-Bulb Temp [C]',
+            '!- Daily Dry-Bulb Temp Range [C]',
+            '!- Dry-Bulb Temp Range Modifier Type',
+            '!- Dry-Bulb Temp Range Modifier Schedule Name',
+            '!- Humidity Condition Type',
+            '!- Wetbulb/Dewpoint at Max Dry-Bulb [C]',
+            '!- Humidity Indicating Day Schedule Name',
+            '!- Humidity Ratio at Maximum Dry-Bulb {kgWater/kgDryAir}',
+            '!- Enthalpy at Maximum Dry-Bulb {J/kg}',
+            '!- Daily Wet-Bulb Temperature Range [deltaC]',
+            '!- Barometric Pressure [Pa]',
+            '!- Wind Speed {m/s}',
+            '!- Wind Direction [Degrees; N=0, S=180]',
+            '!- Rain [Yes/No]',
+            '!- Snow on ground [Yes/No]',
+            '!- Daylight Savings Time Indicator',
+            '!- Solar Model Indicator',
+            '!- Beam Solar Day Schedule Name',
+            '!- Diffuse Solar Day Schedule Name',
+            '!- ASHRAE Clear Sky Optical Depth for Beam Irradiance (taub)',
+            '!- ASHRAE Clear Sky Optical Depth for Diffuse Irradiance (taud)',
+            '!- Clearness [0.0 to 1.2]']
+temp_multipliers = [0.82, 0.88, 0.92, 0.95, 0.98, 1, 0.98, 0.91, 0.74,
+                    0.55, 0.38, 0.23, 0.13, 0.05, 0, 0, 0.06, 0.14, 0.24,
+                    0.39, 0.5, 0.59, 0.68, 0.75]
+
+
+class Ddy(object):
```

Comment:
I thought we said that we were absolutely sure that we are following PEP8 here and naming all of the classes with title case.  The only calss that does not follow this right now is EPW.  The only reason why I haven't changed it is because I know that there should be a simultaneous update of the grasshopper components.  If you agree with me that we should just use title case, then I will update the EPW class and the Grasshopper components in my next pull request.

Reply:
I know we talked about this but looking back I think with acronyms we should keep them uppercase. That will make them easier to read and is closer to what a user would expect. I have seen it in Python libraries. Here is an example:

> class names: CamelCase, with acronyms kept uppercase (HTTPWriter and not HttpWriter)

from this page: https://palletsprojects.com/governance/sourcecode/python-styleguide/